### ISSUE TITLE
Request failed with status code 413

### ISSUE THREAD

**Issue by lublak** [n8n-io/n8n#1447]
**Describe the bug**
I have a lot of data 61357 rows of data with 15 columns.
Currently if i run my workflow it works fine, but if i try to rerun from a single node it gives me an error.
In the ui: Request failed with status code 413
In the console:
```
PayloadTooLargeError: request entity too large
    at readStream (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\raw-body\index.js:155:17)
    at getRawBody (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\raw-body\index.js:108:12)
    at read (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\body-parser\lib\read.js:77:3)
    at jsonParser (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\body-parser\lib\types\json.js:135:5)
    at Layer.handle [as handle_request] (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:317:13)
    at C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:335:12)
    at next (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:275:10)
    at C:\ProgramData\npm\npm\node_modules\n8n\src\Server.ts:353:4
    at Layer.handle [as handle_request] (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:317:13)
    at C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:335:12)
    at next (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:275:10)
    at compression (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\compression\index.js:220:5)
    at Layer.handle [as handle_request] (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:317:13)
    at C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:335:12)
    at next (C:\ProgramData\npm\npm\node_modules\n8n\node_modules\express\lib\router\index.js:275:10)
    at C:\ProgramData\npm\npm\node_modules\n8n\src\Server.ts:342:4
```

**To Reproduce**
Currently impossible to make a small example, i read a lot of csv files.

**Environment (please complete the following information):**
 - OS: Windows 10
 - n8n Version 0.108.0
 - Node.js Version v14.15.1

**Comment by krynble** [comment]
Hi @lublak 

We currently limit the payload that can be sent to n8n to 16MB.

This is to avoid allocating too much memory. When you have this amount of data, it is probably going over this limit.

The manual executions send all data from the interface back to n8n service, where this 16MB limit resides.

If you want to increase this limit for specific use cases, this can be done in the `packages/cli/src/Server.ts` file at this block:

```
		// Support application/json type post data
		this.app.use(bodyParser.json({
			limit: '16mb', verify: (req, res, buf) => {
				// @ts-ignore
				req.rawBody = buf;
			},
		}));
```

For now I don't think this will be changed in our stable versions as increasing this number might degrade performance considerably.

**Comment by lublak** [comment]
@krynble hmm okay. Then I think it should be better to implement a: "do not send all data". Option. 
Currently I manipulate this directly in js.

**Comment by Banbury** [comment]
@krynble How about making this limit configurable per workflow? I regularly handle JSON data much larger than 16MB and my server is perectly capable of handling that.
Hardcoding limits is so 20th century. :)

**Comment by janober** [comment]
It is now possible with configure the max payload size by setting the environment variable `N8N_PAYLOAD_SIZE_MAX`. The value is in MB. So setting:
```bash
export N8N_PAYLOAD_SIZE_MAX=32
```
will set the limit to 32MB.

Will be released with the next version.

**Comment by janober** [comment]
Got released with n8n@0.115.0
